web/telemetry/engine.go

```diff
@@ -46,6 +48,16 @@ type Publisher interface {
 func (e *Engine) Start(ctx context.Context) {
 	log.Infof("Starting Telemetry Engine")
 
+	canPublishTelemetry, err := e.premiumDetection.CanPublishTelemetry()
+	if err != nil {
+		log.Errorf("Unable to start Telemetry Engine. Error: %s", err)
+		return
+	}
+	if !canPublishTelemetry {
+		log.Infof("Telemetry publishing is not supported by this installation")
+		return
+	}
+
```

Comment:
I'm very confused by this. It looks like a one-shot check. Is there anything ensuring that the telemetry engine is started once the `premiumDetection.CanPublishTelemetry()` changes result?

Reply:
good catch. 
Moving the checking step inside the repeat loop.
This way if `premiumDetection.CanPublishTelemetry()` changes result it will be detected and trento will start publishing telemetry.